sprintQuovantis/sprintParam/tests/test_views.py

```diff
@@ -0,0 +1,551 @@
+
+    def test_the_add_sprint_with_blank_sprint_name(self, client):
+        user = User.objects.create_superuser("sunanda",
+                                             password="1234",
+                                             email="ssunanda02@gmail.com")
+        user.save()
+        data = {
+            "username": "sunanda",
+            "password": "1234"
+        }
+        url = reverse("login")
+        response = client.post(url, data)
+        json_data = json.loads(response.content)
+        token = json_data.get('data').get("token")
+        # adding sprint
+        sprint_data = {
+            "sprint_name": "",
+            "entry_date": "2022-02-2 06:00",
+            "expiry_date": "2022-02-22"
+        }
+        url = reverse("add-sprint")
+        header = {
+            "HTTP_AUTHORIZATION": token,
+        }
+        response = client.post(url, sprint_data, **header)
+        assert response.status_code == 400
+
+    def test_the_add_sprint_with_no_token(self, client):
+        user = User.objects.create_superuser("sunanda",
+                                             password="1234",
+                                             email="ssunanda02@gmail.com")
+        user.save()
+        data = {
+            "username": "sunanda",
+            "password": "1234"
+        }
+        url = reverse("login")
+        response = client.post(url, data)
+        json_data = json.loads(response.content)
+        # adding sprint
+        sprint_data = {
+            "sprint_name": "task",
+            "entry_date": "2022-02-2 06:00",
+            "expiry_date": "2022-02-22"
+        }
+        url = reverse("add-sprint")
+        response = client.post(url, sprint_data)
+        assert response.status_code == 400
+
+    def test_fetch_sprint(self, client):
+        user = User.objects.create_superuser("sunanda",
+                                             password="1234",
+                                             email="ssunanda02@gmail.com")
+        user.save()
+        data = {
+            "username": "sunanda",
+            "password": "1234"
+        }
+        url = reverse("login")
+        response = client.post(url, data)
+        json_data = json.loads(response.content)
+        token = json_data.get('data').get("token")
+        # adding sprint
+        sprint_data = {
+            "sprint_name": "task",
+            "entry_date": "2022-02-2 06:00",
+            "expiry_date": "2022-02-22"
+        }
+        url = reverse("add-sprint")
+        header = {
+            "HTTP_AUTHORIZATION": token,
+        }
+        response = client.get(url, sprint_data, **header)
+        assert response.status_code == 200
+
+    def test_fetch_sprint_without_token(self, client):
+        user = User.objects.create_superuser("sunanda",
+                                             password="1234",
+                                             email="ssunanda02@gmail.com")
+        user.save()
```

Comment:
create superuser code is repetitive, common code should be written only once in a method and called everywhere.